examples/opencv_rc.py

```diff
@@ -0,0 +1,541 @@
+
+    def _set_action(self, linear, angular):
+        """
+        Set the directions of Cozmo's movement and send the corresponding
+        command to spin the motors at the right speed.
+        :param linear: Direction. An instance of the Direction enumeration.
+        Either Forward, Backward, or None.
+        :param angular: Direction. An instance of the Direction enumeration.
+        Either Left, Right, or None.
+        :return: None
+        """
+
+        # Check if the new values are different from the old ones before doing
+        # any complex calculations and send motor commands
+        if self._action['linear'] != linear or \
+                self._action['angular'] != angular:
+
+            # Set the value of the new action
+            self._action['linear'] = linear
+            self._action['angular'] = angular
+
+            # If Cozmo is not doing anything, simply stop all the motors
+            if self._action['linear'] == Direction.NONE and \
+                    self._action['angular'] == Direction.NONE:
+                self._cozmo_clt.stop_all_motors()
+
+            # Compute the speed of each wheel depending on Cozmo's action and
+            # the new values for the linear and angular velocities
+            else:
+                # Are we going forward, backward, or not moving in this
+                # direction at all?
+                if self._action['linear'] == Direction.FORWARD:
+                    lin_vel = self._velocity['linear']
+                elif self._action['linear'] == Direction.BACKWARD:
+                    lin_vel = -self._velocity['linear']
+                else:
+                    lin_vel = 0
+
+                # Are we going left, right, or not moving in this direction at
+                # all?
+                if self._action['angular'] == Direction.LEFT:
+                    ang_vel = self._velocity['angular']
+                elif self._action['angular'] == Direction.RIGHT:
+                    ang_vel = -self._velocity['angular']
+                else:
+                    ang_vel = 0
+
+                # Compute the actual speed of each wheel
+                left = min(pc.MAX_WHEEL_SPEED.mmps, lin_vel -
+                           (pc.TRACK_WIDTH.mm * ang_vel) / 2)
+                right = min(pc.MAX_WHEEL_SPEED.mmps, lin_vel +
+                            (pc.TRACK_WIDTH.mm * ang_vel) / 2)
+
+                # Send the command to the motors
+                self._cozmo_clt.drive_wheels(lwheel_speed=left,
+                                             rwheel_speed=right)
+
+    def _stop_all(self, *args):
+        """
+        Simply stop all motors and set both linear and angular actions to NONE.
+        This is a wrapper for ease of use as a callback.
+        :return: None
+        """
+
+        # Set both actions to NONE, which will stop the motors
+        self._set_action(Direction.NONE, Direction.NONE)
+
+    def _on_head_tilt_change(self, value):
```

Comment:
I would suggest sending the command for the new head tilt whenever this changes. Otherwise, it will only be applied when you attempt to move the robot.

Reply:
I am not sure I understand the problem here. Within the `_on_head_tilt_change()` function I assign the new value for the head tilt using `self.head_tilt = value`. This in fact calls the property setter that I defined on line 450. This setter in turn sends an immediate command to Cozmo to move its head to the new angle. Maybe not the most straight forward way of doing this, but it works.

Reply:
😱 you are right! I completely missed the setters! good work with that, keep it!